### ISSUE TITLE
Client Biome: connection to server is erroring. Cannot call write after a stream was destroyed

### ISSUE THREAD

**Issue by huseeiin** [biomejs/biome-vscode#86]
```
read ECONNRESET
Shutting down server.
[Error - 9:28:13 AM] Client Biome: connection to server is erroring.
Cannot call write after a stream was destroyed
[Error - 9:28:13 AM] Stopping server failed
  Message: Cannot call write after a stream was destroyed
  Code: -32099 
```

**Comment by nhedger** [comment]
Hey @huseeiin,

Thanks for opening this issue.

There isn't much for me to go on here, so I'll close this for now.

Please reopen with a reproduction and a detailed explanation of what happens.

**Comment by kauly** [comment]
This is constantly happening here on VS Code version 1.89.1 on Ubuntu.

VS Code version

```
Version: 1.89.1
Commit: dc96b837cf6bb4af9cd736aa3af08cf8279f7685
Date: 2024-05-07T05:16:23.416Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Linux x64 6.5.0-35-generic snap
```

Biome console outuput:
```bash
[cli-stdout] data 44
[cli] exit 0
Connecting to "/home/kauly/.cache/biome/biome-socket-1.7.2" ...
[cli-stderr] data 395
[Error - 4:24:56 PM] Client Biome: connection to server is erroring. Shutting down server.
[Error - 4:24:56 PM] Client Biome: connection to server is erroring. Shutting down server.
[Error - 4:24:56 PM] Stopping server failed
  Message: Cannot call write after a stream was destroyed
  Code: -32099 
[Error - 4:24:56 PM] Stopping server failed
  Message: Cannot call write after a stream was destroyed
  Code: -32099 
[Error - 4:24:56 PM] Stopping server failed
  Message: Cannot call write after a stream was destroyed
  Code: -32099 
[cli-stderr] end
[cli-stderr] finish
[cli-stdout] end
[cli-stdout] finish
[cli-stdout] close
[cli] close 0
[cli-stderr] close
```

**Comment by ematipico** [comment]
@kauly 

I checked a couple of answers on the internet.

Can you make sure that `biome-logs` have writing access? https://biomejs.dev/guides/integrate-in-editor/#daemon-logs

**Comment by kauly** [comment]
@ematipico thank you for your response. By `biome-logs` do you meaning the folder? I have the folder with many logs in it so I guess the permissions are ok.

```shell
-rw-rw-r-- 1 kauly kauly 2901113 mai 19 17:17 server.log.2024-05-19-20
-rw-rw-r-- 1 kauly kauly 2002201 mai 19 21:54 server.log.2024-05-20-00
-rw-rw-r-- 1 kauly kauly 1188011 mai 19 22:03 server.log.2024-05-20-01
-rw-rw-r-- 1 kauly kauly   45990 mai 20 11:01 server.log.2024-05-20-14
```

**Comment by kauly** [comment]
This is a log where the error happens.
[server.log.2024-05-20-14.txt](https://github.com/biomejs/biome-vscode/files/15378556/server.log.2024-05-20-14.txt)


```
│ │ │ ├─ 17ms DEBUG biome_rowan::ast::batch pushing change...
│ │ │ ├─ 19ms ERROR biome_cli::panic Biome encountered an unexpected error
│ │ │ │ 
│ │ │ │ This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:
│ │ │ │ 
│ │ │ │ Source Location: crates/biome_js_semantic/src/semantic_model/scope.rs:115:33
│ │ │ │ Thread Name: tokio-runtime-worker
│ │ │ │ Message: no entry found for key
│ │ │ │ 
│ │ ├─┘
│ ├─┘
├─┘
INFO biome_cli::commands::daemon Received shutdown signal
````

**Comment by kauly** [comment]
ok @Sec-ant when it happens again I will do that.

**Comment by ematipico** [comment]
> This is a log where the error happens. [server.log.2024-05-20-14.txt](https://github.com/biomejs/biome-vscode/files/15378556/server.log.2024-05-20-14.txt)
> 
> ```
> │ │ │ ├─ 17ms DEBUG biome_rowan::ast::batch pushing change...
> │ │ │ ├─ 19ms ERROR biome_cli::panic Biome encountered an unexpected error
> │ │ │ │ 
> │ │ │ │ This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to [https://github.com/biomejs/biome/issues/](https://github.com/biomejs/biome/issues/?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) along with the following information to help us fixing the issue:
> │ │ │ │ 
> │ │ │ │ Source Location: crates/biome_js_semantic/src/semantic_model/scope.rs:115:33
> │ │ │ │ Thread Name: tokio-runtime-worker
> │ │ │ │ Message: no entry found for key
> │ │ │ │ 
> │ │ ├─┘
> │ ├─┘
> ├─┘
> INFO biome_cli::commands::daemon Received shutdown signal
> ```

We fixed this issue in Biome v1.7.3.

**Comment by Sec-ant** [comment]
> We fixed this issue in Biome v1.7.3.

@ematipico You have better memories me 😄, I just realized the file and line number looks familiar...
